src/sentry/testutils/cases.py

```diff
@@ -1914,7 +1915,8 @@ def setUp(self, provider="dummy"):
         self.auth_provider = AuthProviderModel(organization=self.organization, provider=provider)
         self.auth_provider.enable_scim(self.user)
         self.auth_provider.save()
-        self.login_as(user=self.user)
+        self.scim_user = ApiToken.objects.get(token=self.auth_provider.get_scim_token()).user
```

Comment:
This feels ugly. Is there a better way to get the auth_provider user?

Reply:
Not really. Ideally we could do something like
```python
(self.scim_user,) = ApiToken.objects.filter(token=self.auth_provider.get_scim_token()).values_list("user", flat=True)
```
but that just returns the foreign key, not the `User` object. There are a few other ways to slice it but, considering that we don't need to optimize performance for `testutils`, I think what you've written is the cleanest option.

Reply:
you could make a helper method that uses the token directly e.g. https://github.com/getsentry/sentry/blob/509241d00e459b2509c12d2d4a92bbca03a4873e/tests/sentry/api/endpoints/test_project_grouping_configs.py#L19